cloudinit/sources/DataSourceOpenStack.py

```diff
@@ -73,6 +73,12 @@ def __str__(self):
         return mstr
 
     def wait_for_metadata_service(self):
+        DEF_MD_URLS = [
+            "http://[fe80::a9fe:a9fe%25{iface}]".format(
+                iface=net.find_fallback_nic()
```

Comment:
The datasource object already has a reference to the fallback interface through the distro object (`self.distro.fallback_interface`). Using that would be preferred instead of a redundant call to `net.find_fallback_nic`.

Reply:
@holmanb I pushed a new revision with this change an also proper URL-encoding of the `%` to `%25`.

PTAL.